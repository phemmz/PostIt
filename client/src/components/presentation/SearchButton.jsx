import React from 'react';
import { Link } from 'react-router';
import ReactTooltip from 'react-tooltip';

/**
 * @description SearchButton Component
 * @return {*} div
 */
const SearchButton = () => {
  return (
    <div className="center">
      <ReactTooltip />
      <Link
        to="/search"
        className="searchbtn waves-effect waves-light btn center search-users"
        data-tip="Search Users"
      >Search Users</Link>
    </div>
  );
};

export default SearchButton;
