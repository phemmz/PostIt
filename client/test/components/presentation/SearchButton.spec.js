import React from "react";
import { mount } from "enzyme";
import SearchButton from '../../../src/components/presentation/SearchButton';

describe("SearchButton Component", () => {
  let mountedSearchButton;

  const searchButton = () => {
    if (!mountedSearchButton) {
      mountedSearchButton = mount(
        <SearchButton />
      );
    }
    return mountedSearchButton;
  }

  beforeEach(() => {
    searchButton: undefined;
  });

  it("should always render a div", () => {
    const divs = searchButton().find("div");
    expect(searchButton()).toMatchSnapshot();
    expect(divs.length).toBeGreaterThan(0);
  });

  it("should contain everything else that gets rendered", () => {
    const divs = searchButton().find("div");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(searchButton().children());
  });
});
