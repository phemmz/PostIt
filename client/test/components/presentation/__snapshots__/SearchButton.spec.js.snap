// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton Component should always render a div 1`] = `
<SearchButton>
  <div
    className="center"
  >
    <ReactTooltip
      insecure={true}
      resizeHide={true}
      wrapper="div"
    >
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip"
      />
    </ReactTooltip>
    <Link
      className="searchbtn waves-effect waves-light btn center search-users"
      data-tip="Search Users"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/search"
    >
      <a
        className="searchbtn waves-effect waves-light btn center search-users"
        data-tip="Search Users"
        onClick={[Function]}
        style={Object {}}
      >
        Search Users
      </a>
    </Link>
  </div>
</SearchButton>
`;
